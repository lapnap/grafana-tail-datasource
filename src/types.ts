import {DataQuery, DataSourceJsonData} from '@grafana/ui';

export interface TailQuery extends DataQuery {
  path?: string;
}

export interface TailOptions extends DataSourceJsonData {
  // The datasource options
}
