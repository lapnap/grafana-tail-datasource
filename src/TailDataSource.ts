// Types
import {
  DataQueryRequest,
  DataQueryResponse,
  DataSourceApi,
  DataSourceInstanceSettings,
} from '@grafana/ui';

import {TailQuery, TailOptions} from './types';

export class TailDataSource extends DataSourceApi<TailQuery, TailOptions> {
  constructor(instanceSettings: DataSourceInstanceSettings<TailOptions>) {
    super(instanceSettings);
  }

  /**
   * Convert a query to a simple text string
   */
  getQueryDisplayText(query: TailQuery) {
    return `Get Data: ${query.path}`;
  }

  query(options: DataQueryRequest<TailQuery>): Promise<DataQueryResponse> {
    // if (!this.settings) {
    //   return Promise.reject('no settings');
    // }

    // const {url, jsonData} = this.settings;

    // console.log('FETCH', options, url, jsonData);

    return Promise.resolve({data: []});
  }

  testDatasource() {
    return new Promise((resolve, reject) => {
      resolve({
        status: 'success',
        message: 'Yes',
      });
    });
  }
}
