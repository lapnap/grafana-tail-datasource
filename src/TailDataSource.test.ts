import {TailDataSource} from './TailDataSource';
import {TailOptions} from './types';
import {readCSV, DataSourceInstanceSettings, PluginMeta, DataQueryRequest} from '@grafana/ui';

describe('InputDatasource', () => {
  const data = readCSV('a,b,c\n1,2,3\n4,5,6');
  const instanceSettings: DataSourceInstanceSettings<TailOptions> = {
    id: 1,
    type: 'x',
    name: 'xxx',
    url: 'hello',
    meta: {} as PluginMeta,
    jsonData: {},
  };

  describe('when querying', () => {
    test('should return the saved data with a query', () => {
      const ds = new TailDataSource(instanceSettings);
      const options = {
        targets: [{refId: 'Z'}],
      } as DataQueryRequest;

      return ds.query(options).then(rsp => {
        expect(rsp.data.length).toBe(1);

        const series = rsp.data[0];
        expect(series.refId).toBe('Z');
        expect(series.rows).toEqual(data[0].rows);
      });
    });
  });
});
