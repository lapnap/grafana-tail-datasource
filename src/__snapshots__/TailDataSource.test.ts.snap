// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TailDatasource when querying should give error with no path defined 1`] = `"Missing Path"`;
